pkg/objstore/s3/s3.go

```diff
@@ -256,22 +263,40 @@ func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
 	return true, nil
 }
 
+func (b *Bucket) guessFileSize(name string, r io.Reader) int64 {
```

Comment:
is this a guess, we either get the size or `-1`

Reply:
It is because we try to guess implementation of interface which is a hacky guess.